### PR TITLE
Cadence

### PR DESCRIPTION
**Pull request by TheLillyleaf**
Implement cadence in model
New functions to calculate balance and spend

### CODE REVIEW

File: Receipt/Total.ts [hunk 1]
```diff
@@ -11,6 +10,10 @@ export namespace Total {
 	export const is = type.is
 	export const flaw = type.flaw
 	export function validate(total: Total, currency: isoly.Currency): boolean {
-		return currency == total.net[1] && total.net[1] == total.vat[1]
+		return currency == total.net.currency && currency == total.vat.currency
+	}
+	// missing tests. maybe use in purchase? unused.
```
Comment (EliasEriksson) [inline Receipt/Total.ts:15]: is this tested?

File: Receipt/index.ts [hunk 2]
```diff
@@ -82,12 +79,19 @@ export namespace Receipt {
 		}
 		return Array.from(list(roots))
 	}
-	export function validate(receipt: Receipt, currency?: isoly.Currency): boolean {
-		return (
-			!!receipt.id &&
-			!!receipt.original &&
-			receipt.date < isoly.DateTime.now() &&
-			(!receipt.total.length || !currency || receipt.total.every(total => Total.validate(total, currency)))
+	// maybe expand on this to shorten purchase validate
```
Comment (EliasEriksson) [inline Receipt/index.ts:82]: is this comment resolved?

File: Report/Expense/Creatable.ts [hunk 3]
```diff
@@ -1,14 +1,17 @@
+import { userwidgets } from "@userwidgets/model"
 import { isly } from "isly"
+import type { CostCenter } from "../../CostCenter"
+import { Identifier } from "../../CostCenter/Identifier"
 
 export interface Creatable {
-	costCenterIds: string[]
-	emails?: string[]
+	costCenters: CostCenter["id"][]
```
Comment (EliasEriksson) [inline Report/Expense/Creatable.ts:7]: CostCenter.Identifier[]

File: Report/Expense/Preview.ts [hunk 4]
```diff
@@ -1,18 +1,18 @@
 import { isoly } from "isoly"
-import { Organization } from "@userwidgets/model"
+import { userwidgets } from "@userwidgets/model"
 import { isly } from "isly"
 import { Purchase } from "../../Purchase"
 
 export interface Preview {
 	userExpenses: Record<Purchase["buyer"], Purchase[] | undefined>
-	organization: Organization["id"]
+	organization: userwidgets.Organization["id"]
```
Comment (EliasEriksson) [inline Report/Expense/Preview.ts:8]: userwidgets.Organization.Identifier

File: Report/Receipt/Creatable.ts [hunk 5]
```diff
@@ -1,12 +1,14 @@
 import { isly } from "isly"
+import type { CostCenter } from "../../CostCenter"
+import { Identifier } from "../../CostCenter/Identifier"
 
 export interface Creatable {
-	costCenterIds: string[]
+	costCenters: CostCenter["id"][]
```
Comment (EliasEriksson) [inline Report/Receipt/Creatable.ts:6]: CostCenter.Identifier[]

File: Report/Receipt/Preview.ts [hunk 6]
```diff
@@ -7,9 +8,10 @@ import { Receipt } from "../../Receipt"
 export interface Preview {
 	costCenters: {
 		costCenter: CostCenter
+		// change to: purchases: Purchase[] and filter out the irrelevant receipts from its array instead
 		receipts: { receipt: Receipt; purchase: Purchase }[]
 	}[]
-	organization: string
+	organization: userwidgets.Organization["id"]
```
Comment (EliasEriksson) [inline Report/Receipt/Preview.ts:14]: userwidgets.Organization.Identifier

File: Transaction/index.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import { isly } from "isly"
-import { Delegation } from "../Delegation"
-import { Purchase } from "../Purchase"
+// import { Delegation } from "../Delegation"
+// import { Purchase } from "../Purchase"
```
Comment (EliasEriksson) [inline Transaction/index.ts:3]: remove comment

**Review by EliasEriksson — CHANGES_REQUESTED** [review]

File: Report/Expense/Preview.ts (hunk 4 above)
Comment (TheLillyleaf) [reply Report/Expense/Preview.ts:8]: Was supposed to be the name, but good catch regardless

**Review by EliasEriksson — APPROVED** [review]

**Review by EliasEriksson — APPROVED** [review]

**Pull request merged**